File: Lecture008 Switch Statement & Functions/primeNumber.cpp
#include<iostream>
using namespace std;
// 0 -> Not a Prime no.
// 1 -> Prime no.

bool isPrime(int n ) {
   for(int i = 2; i < n; i++){
       if(n % i == 0) {
       //divide hogya h , not a prime no.
          return 0;
        }
   }
   return 1;
}

int main() {
   
   int n ;
   cin >> n;
   if(isPrime(n)) {
        cout <<" is a Prime number " <<endl;
   }
   else{
       cout<< "Not a prime number "<<endl;

   }
}

File: Lecture024 Maths for DSA/PrimeNumber.java
package MathsForDSA;
import java.util.Scanner;

public class PrimeNumber {
    // Check Whether a Given Number is Prime or Not
    public static void main(String [] args){
        // Creating Scanner Object for taking inputs from the USER
        Scanner sc = new Scanner(System.in);
        // Take any Number as a Input from the USER
        System.out.print(" Enter Any Number: ");
        int num = sc.nextInt();
        // Display is the Number Entered by USER is Prime or NOT
        System.out.println(num + " is: " + isPrime(num));
    }

    static String isPrime(int num){
        if(num == 0)
            return " Neither Prime Nor Composite ";
        if( num < 0 )
            return " Not Prime ";
        for(int i=2; i*i < num; i++){
            if(num % i == 0)
                return " Not Prime ";
        }
        return " Prime ";
    }
}
